File: lang-js/No2824.js
/**
 * 统计和小于目标的下标对数目
 * @param nums
 * @param target
 * @returns {number}
 */
let countPairs = (nums, target) => {
    nums.sort((o1, o2) => o1 - o2);
    let sum = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] + nums[j] < target) {
                sum++;
            } else {
                break;
            }
        }
    }
    return sum;
};

File: lang-py/No2824.py
"""
  author <a href="https://github.com/Matthew-Han">Matthew Han</a>
  since 2024/1/2 10:58:00
"""
from typing import List

class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        ans = 0
        nums.sort()
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] < target:
                    ans += 1
                else:
                    break
        return ans

